sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/implementation/AzureEnvironment.java

```diff
@@ -0,0 +1,352 @@
+     */
+    public String getSqlManagementEndpoint() {
+        return endpoints.get("sqlManagementEndpointUrl");
+    }
+
+    /**
+     * @return the dns suffix for sql servers.
+     */
+    public String getSqlServerHostnameSuffix() {
+        return endpoints.get("sqlServerHostnameSuffix");
+    }
+
+    /**
+     * @return the Active Directory login endpoint.
+     */
+    public String getActiveDirectoryEndpoint() {
+        return endpoints.get("activeDirectoryEndpointUrl").replaceAll("/$", "") + "/";
+    }
+
+    /**
+     * @return The resource ID to obtain AD tokens for.
+     */
+    public String getActiveDirectoryResourceId() {
+        return endpoints.get("activeDirectoryResourceId");
+    }
+
+    /**
+     * @return the template gallery endpoint.
+     */
+    public String getGalleryEndpoint() {
+        return endpoints.get("galleryEndpointUrl");
+    }
+
+    /**
+     * @return the Active Directory resource ID.
+     */
+    public String getGraphEndpoint() {
+        return endpoints.get("activeDirectoryGraphResourceId");
+    }
+
+    /**
+     * @return the Microsoft Graph resource ID.
+     */
+    public String getMicrosoftGraphEndpoint() {
+        return endpoints.get("microsoftGraphResourceId");
+    }
+
+    /**
+     * @return the Data Lake resource ID.
+     */
+    public String getDataLakeEndpointResourceId() {
+        return endpoints.get("dataLakeEndpointResourceId");
+    }
+
+    /**
+     * @return the Active Directory api version.
+     */
+    public String getActiveDirectoryGraphApiVersion() {
+        return endpoints.get("activeDirectoryGraphApiVersion");
+    }
+
+    /**
+     * @return the endpoint suffix for storage accounts.
+     */
+    public String getStorageEndpointSuffix() {
+        return endpoints.get("storageEndpointSuffix");
+    }
+
+    /**
+     * @return the keyvault service dns suffix.
+     */
+    public String getKeyVaultDnsSuffix() {
+        return endpoints.get("keyVaultDnsSuffix");
+    }
+
+    /**
+     * @return the data lake store filesystem service dns suffix.
+     */
+    public String getAzureDataLakeStoreFileSystemEndpointSuffix() {
+        return endpoints.get("azureDataLakeStoreFileSystemEndpointSuffix");
+    }
+
+    /**
+     * @return the data lake analytics job and catalog service dns suffix.
+     */
+    public String getAzureDataLakeAnalyticsCatalogAndJobEndpointSuffix() {
+        return endpoints.get("azureDataLakeAnalyticsCatalogAndJobEndpointSuffix");
+    }
+
+    /**
+     * @return the log analytics endpoint.
+     */
+    public String getLogAnalyticsEndpoint() {
+        return endpoints.get("azureLogAnalyticsResourceId");
+    }
+
+    /**
+     * @return the log analytics endpoint.
+     */
+    public String getApplicationInsightsEndpoint() {
+        return endpoints.get("azureApplicationInsightsResourceId");
+    }
+
+    /**
+     * The enum representing available endpoints in an environment.
+     */
+    public enum Endpoint {
```

Comment:
Should this be an `ExpandableStringEnum`? Or at least could the map use this as the key so we don't need to maintain the same string identifier in multiple locations (reducing copy and paste errors).

Reply:
refactored the strings into constants.